File: core/src/main/java/javabot/operations/WeatherOperation.java
package javabot.operations;

import com.antwerkz.sofia.Sofia;
import javabot.Message;
import javabot.dao.impl.WeatherDao;
import javabot.model.Weather;

import javax.inject.Inject;

/**
 * Gets current weather conditions for a place given as a parameter.
 */
public class WeatherOperation extends BotOperation {
    @Inject
    private WeatherDao weatherDao;

    @Override
    public boolean handleMessage(final Message event) {
        final String message = event.getValue();
        if (message.toLowerCase().startsWith("weather ")) {
            final String place = message.substring("weather ".length()).trim();
            final Weather result = weatherDao.getWeatherFor(place);
            if (result == null) {
                getBot().postMessage(event.getChannel(), event.getUser(), Sofia.weatherUnknown(place), event.isTell());
            } else {
                getBot().postMessage(event.getChannel(), event.getUser(), result.toString(), event.isTell());
            }
            return true;
        }
        return false;
    }

}

File: src/main/kotlin/javabot/operations/WeatherOperation.kt
package javabot.operations

import com.antwerkz.sofia.Sofia
import java.util.Locale
import javabot.Javabot
import javabot.Message
import javabot.dao.AdminDao
import javabot.dao.weather.WeatherDao
import javax.inject.Inject

/** Gets current weather conditions for a place given as a parameter. */
class WeatherOperation
@Inject
constructor(bot: Javabot, adminDao: AdminDao, var weatherDao: WeatherDao) :
    BotOperation(bot, adminDao) {
    override fun handleMessage(event: Message): List<Message> {
        val responses = arrayListOf<Message>()
        val message = event.value
        if (message.lowercase(Locale.getDefault()).startsWith("weather ")) {
            val place = message.substring("weather ".length).trim()
            val result = weatherDao.getWeatherFor(place)
            responses.add(Message(event, result?.toString() ?: Sofia.weatherUnknown(place)))
        }
        return responses
    }
}
